### PR TITLE
First stable release of the VenomPrecision Application

### LINKED ISSUE #32: First stable release
Release VenomPrecision 1.0

### PR DESCRIPTION
**Pull request by limavfabio**
This pull request introduces the first stable release of the VenomPrecision application, which includes a range of new features, enhancements, and bug fixes. The application is built using React and Redux, and the development environment utilizes Vite.

Key Features:

- Login and Signup Features: Users can now create accounts, log in, and securely authenticate themselves within the VenomPrecision application.

- Product Creation and Management: Users have the ability to create new products and easily manage them within the application. They can also reserve products, specifying the desired city and date.

- Fully Responsive Design: The application is designed to be fully responsive, offering seamless user experiences across various devices, including desktop and mobile.

- Deployment on Netlify: The application has been successfully deployed on Netlify and is accessible at the following URL: https://velvety-genie-9ab88d.netlify.app/. Users can now access the application and explore its functionalities.

Additional Features:

- Toast Notifications: Toast notifications have been implemented to provide users with important updates, alerts, and messages in a non-intrusive manner.

- Loading Spinners: Loading spinners have been added to indicate the progress of certain actions or processes within the application, ensuring a smoother user experience.

- Responsive Header and Sidebar: The application now features a responsive header and sidebar, adapting seamlessly to different screen sizes and improving navigation and accessibility.

- Carousel for Main Page Items: A carousel component has been incorporated into the main page, allowing users to browse through a visually appealing presentation of all available items.

You can find the Pull Request for the back-end project here: https://github.com/limavfabio/VenomPrecision-back-end/pull/25

### CODE REVIEW

File: src/components/Item-form/ItemForm.jsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+              </button>
+              <button
+                type="button"
+                className="ml-5 mt-4 bg-[#E4E5E9] px-6 py-2 text-black hover:bg-[#cacacc] focus:bg-[#9a9a9c]"
+                onClick={() => redirect(-1)}
+              >
+                Cancel
+              </button>
+            </div>
+          </form>
```
Comment (mcrd25) [inline src/components/Item-form/ItemForm.jsx]: Some of these elements are reptitive. Don't forget DRY.

Consider using a subcomponent for form inputs like so:

```js
const FormInput = ({ id, value, placeholder, onChange, className, labelText }) => (
  <label htmlFor={id} className="mb-2 block">
    <p className="font-semibold">{labelText}</p>
    <input
      type="text"
      id={id}
      value={value}
      placeholder={placeholder}
      onChange={onChange}
      className={className}
      required
    />
  </label>
);

FormInput.propTypes = {
  id: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  className: PropTypes.string.isRequired,
  labelText: PropTypes.string.isRequired,
};
```

**Review by mcrd25 — CHANGES_REQUESTED** [review]
Hi Team,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- All functional requirements seem to have been met 👍 
- README is good 👍 
- Deployed live demo 👍 

## Required Changes ♻️
_Check the comments under the review._
- *  I am not seeing tests for your Front-End code - this is something we expect to be implemented for the final evaluation. Here are some resources you can use:
   *  https://reactjs.org/docs/testing.html
   * https://jestjs.io/docs/tutorial-react
   * https://www.freecodecamp.org/news/testing-react-hooks/
- Don't forget that the application needs to look exactly like design with exception of photos, titles and descriptions

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- I noticed you started with jest, please do consider adding tests using jest
- There are some candidates to make some of these html components reusable.
- Seeing where you can make code better by following best practices: See the following articles
   - https://www.freecodecamp.org/news/best-practices-for-react/
   - https://betterprogramming.pub/21-best-practices-for-a-clean-react-project-df788a682fb

**Hope all is well, and that my comments were helpful. 😸**

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by JakariaJishan** [comment]
@mcrd25 
hello mcrd25,
thanks for your suggestions,  could you provide some notable examples of which part of our website differs from the original design?

**Review by Meltrust — APPROVED** [review]
#### Hi @limavfabio,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Tests have been implemented✔️ 
- console.log statements have been removed✔️ 
- ItemForm.jsx has been refactored✔️ 
- After refactoring, application still works ✔️ 
- Good job, and good luck ✔️ 



Cheers and Happy coding!👏👏👏


------

**Pull request merged**